### DISCUSSION TITLE
Client Routing not working on dynamically generated links

### DISCUSSION THREAD

**Discussion by brillout** [vikejs/vike#646] [Discussion] [answered]
Moved from Discord:

> hey quick question regarding client side navigation - as far as I understand, clicks on `<a>` tags are intercepted so as to prevent a full reload. This works most of the time for me - but in a search component where links are generated dynamically, clicking any link leads to a full page reload, which is not desired. Does anyone have any pointers as to what may cause this? I do not have any special attributes set on the anchor tag.

**Comment by brillout** [comment]
It should work as VPS intercepts clicks globally:
https://github.com/brillout/vite-plugin-ssr/blob/fd7a06b0091ba50d83b53e84e93975c2e2f1b554/vite-plugin-ssr/client/router/useClientRouter.ts#L247
https://github.com/brillout/vite-plugin-ssr/blob/fd7a06b0091ba50d83b53e84e93975c2e2f1b554/vite-plugin-ssr/client/router/useClientRouter.ts#L274-L284

You can insert a couple of debug `console.log()`s in `node_modules/vite-plugin-ssr/dist/esm/client/router/useClientRouter.js` to see what's going on. (You may need to clear Vite's cache `$ vite --force` / `$ rm -rf node_modules/.vite/`.)

**Comment by 4350pChris** [comment, marked as answer]
Just a follow up - the problem was that the nodes didn't have parent elements for some reason. I didn't figure out why that was the case, but basically what happened was that the node the click handler got was not a node that was in the DOM, but a virtual node by Vue that corresponded to a component file. I ended up removing that component and lifting it into the parent component, at which point the click handler got the correct node and was able to traverse the DOM to find the parent <a> tag. I'm baffled how this could happen.
